drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1564,10 +1564,8 @@ static void get_mac_addr_from_i2c_eeprom(u8_t mac_addr[6])
 		return;
 	}
 
-	i2c_burst_read_addr(dev, CONFIG_ETH_SAM_GMAC_MAC_I2C_SLAVE_ADDRESS,
-			    (u8_t *)&iaddr,
-			    CONFIG_ETH_SAM_GMAC_MAC_I2C_INT_ADDRESS_SIZE,
-			    mac_addr, 6);
+	i2c_write_read(dev, &iaddr, CONFIG_ETH_SAM_GMAC_MAC_I2C_INT_ADDRESS_SIZE,
```

Comment:
Yes, it would be better in the commit that removes the deprecated implementations, or as a separate commit.  I'll queue one or the other for the next push.